koku/api/report/aws/query_handler.py

```diff
@@ -213,6 +215,136 @@ def _build_sum(self, query, annotations):
             self._pack_data_object(query_sum, **self._mapper.PACK_DEFINITIONS)
         return query_sum
 
+    def _get_associated_tags(self, query_table, base_query_filters):  # noqa: C901
+        """
+        Query the reporting_awscostentrylineitem_daily_summary for existence of associated
+        tags grouped by the account.
+
+        Args:
+            query_table (django.db.model) : Table containing the data against which we want to check for tags
+            base_query_filters (django.db.model.Q) : Query filters to apply to table arg
+
+        Returns:
+            dict : {account (str): tags_exist(bool)}
+        """
+
+        def __resolve_op(op_str):
+            """
+            Resolve a django lookup op to a PostgreSQL operator
+
+            Args:
+                op_str (str) : django field lookup operator string
+
+            Returns:
+                str : Translated PostgreSQL operator
+            """
+            op_map = {
+                None: "=",
+                "": "=",
+                "lt": "<",
+                "lte": "<=",
+                "gt": ">",
+                "gte": ">=",
+                "contains": "like",
+                "icontains": "ilike",
+                "startswith": "like",
+                "istartswith": "ilike",
+                "endswith": "like",
+                "iendswith": "ilike",
+            }
```

Comment:
Your operator map is different in a couple places than how it's implemented in the Django Postgres backend. ( See: https://github.com/django/django/blob/master/django/db/backends/postgresql/base.py#L106-L121 )

Do you have an opinion on the differences between, for example, `ILIKE` versus `LIKE UPPER(%s)`?

Secondarily, do you think it's worthwhile to use the `DatabaseWrapper.operators` from Django's postgres backend instead of implementing our own version of it? It seems like it'll lower the maintenance overhead to use Django's version.

Reply:
I know it's a bit different. In this case, at this time, both ILIKE and UPPER(col) LIKE UPPER(val) are both going to break indexing equally. 

I'm working on means to fix this on the ocp query performance ticket. Some significant work there should translate to AWS, Azure, etc.